src/Tribe/Views/V2/Theme_Compatibility.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Add theme compatibility things here.
+ * @TODO: This is an implementation to set a body class we can use in the common implementation.
+ * @TODO: Check this once we move forward with the views structure.
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2
+ */
+namespace Tribe\Events\Views\V2;
+
+use Tribe\Events\Views\V2\Template_Bootstrap;
+
+class Theme_Compatibility {
+
+	/*
+	 * List of themes which have compatibility fixes
+	 */
+	public $themes_with_compatibility_fixes = [
+		'avada',
+		'divi',
+		'enfold',
+		'genesis',
+		'twentyseventeen',
+		'twentynineteen',
+	];
+
+
+	/**
+	 * Register
+	 *
+	 * @since  TBD
+	 *
+	 *
+	 * @return void
+	 */
+	public function register() {}
+
+	/**
+	 * Checks if theme needs a compatibility fix.
+	 *
+	 * @param string $theme Name of template from WP_Theme->Template, defaults to current active template.
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed
+	 */
+	public function needs_compatibility_fix( $theme = null ) {
+		// Defaults to current active theme
+		if ( $theme === null ) {
+			$theme = get_stylesheet();
+		}
+
+		/**
+		 * Allows to filter the theme list with compatibility fixes.
+		 *
+		 * @since  TBD
+		 *
+		 * @param  array  $themes_with_compatibility_fixes A list of themes we provide compatibility for.
+		 */
+		$theme_compatibility_list = apply_filters( 'tribe_events_views_v2_themes_compatibility_fixes', $this->themes_with_compatibility_fixes );
+
+		return in_array( $theme, $theme_compatibility_list );
+	}
+
+	/**
+	 * Add the theme to the body class.
+	 *
+	 * @since TBD
+	 *
+	 * @return array $classes
+	 */
+	public function body_class( $classes ) {
+
+		if (
+			! tribe( Template_Bootstrap::class )->should_load()
+			|| ! $this->needs_compatibility_fix()
+		) {
+			return $classes;
+		}
+
+		$child_theme  = get_option( 'stylesheet' );
+		$parent_theme = get_option( 'template' );
+
+		// if the 2 options are the same, then there is no child theme.
+		if ( $child_theme == $parent_theme ) {
+			$child_theme = false;
+		}
+
+		if ( $child_theme ) {
+			$theme_classes = "tribe-theme-parent-$parent_theme tribe-theme-child-$child_theme";
```

Comment:
From a CSS point of view, is there a way we can target only one class, rather than looking for `.tribe-theme-parent-twentyseventeen`, `.tribe-theme-child-twentyseventeen`, and `.tribe-theme-twentyseventeen`, for example? This will make the theme overrides 3x as large.

Reply:
Sure! I was replicating what we currently have right now (classes). Maybe instead of parent we set the `tribe-theme-twentyseventeen` there and add the child theme class just in case. Agree?

Reply:
Sure yeah, I wonder what the reason was for the decision previously. If it was a valid reason we may need to keep it or think of something to cover that case.